tests/test_url_shortening.py

```diff
@@ -0,0 +1,206 @@
+"""Testing all things shortening of long urls."""
+import json
+import unittest
+from app import create_app, db
+from app.models import User
+from base64 import b64encode
+from flask import url_for
+
+
+class ApiShorten(unittest.TestCase):
+    """Test shortening of urls."""
+
+    def setUp(self):
+        """Setup app for testing before each test case."""
+        self.app = create_app('testing')
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+        db.create_all()
+        self.client = self.app.test_client(use_cookies=True)
+        self.user = User(first_name='ichiato', last_name='ikikin',
+                         email='ichiato@yahoo.com', password='password')
+        self.post_data = json.dumps({
+            "url": "http://www.andela.com",
+            })
+        self.post_data2 = json.dumps({})
+        self.post_data3 = json.dumps({
+            "url": "http://www.andela.com",
+            "vanity_string": "python"
+            })
+
+    def tearDown(self):
+        """Delete app and db instances after each test case."""
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
+
+    def api_email_auth_headers(self, email, password):
+        """Setup an authorization header with an email and password."""
+        return {
+            'Authorization': 'Basic ' + b64encode(
+                (email + ':' + password).encode('utf-8')).decode('utf-8'),
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+
+    def api_token_auth_headers(self, token):
+        """Setup an authorization header with a token."""
+        return {
+            'Authorization': 'Basic ' + b64encode(
+                (token + ':' + '').encode('utf-8')).decode('utf-8'),
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
```

Comment:
The comment about Base class would also include these methods or you could have a testing_utils file to house these.